Make kickstart rescue command noninteractive.

rvykydal opened the pull request:
It should decide on what to do (mount root, mount root readonly) just
based on rescue command options (in case only one system root is found,
otherwise it goes interactive asking for which root to mount).
This would make repairing systems via kickstart %pre and %post sections
automatically without tmux hacks possible.

This is quite dirty/pragmatic solution, I intend to rewrite rescue to remove more execution logic out of the tui spoke, perhaps in scope of moving the ui to simpleline.

jkonecny12 commented:
Please fix this error:

```
************* Module ../pyanaconda/rescue.py
W0107(unnecessary-pass):143,16: RescueMode.initialize: Unnecessary pass statement
FAIL pylint/runpylint.py (exit status: 4)
```

rvykydal commented:
@jkonecny12 I've addressed you remarks and added two more patches separating UI and execution logic. The first one is more about shuffling things. As for the execution functions I've tried to clean them up a bit in the second additional patch. There is still some room for cleanup of functions I haven't touched by the patches, but that was not my point in this set.
Also the prompt/input (mis)usage is still quite wierd in the spokes. Next step would be to move the UI to the new simpleline while sanitizing it.

rvykydal commented:
Adding also Read The Docs documentation for rescue.

rvykydal commented:
jenkins, test this please

jkonecny12 commented on pyanaconda/rescue.py:
```diff
@@ -37,11 +37,13 @@
 import os
```
The commit comment ``... would be to make the UI "will" more sane ..`` is confusing. I guess the `will` should not be there.

rvykydal replied:
Yes, thanks.

jkonecny12 commented on pyanaconda/rescue.py:
```diff
@@ -115,41 +96,275 @@ def makeResolvConf(instPath):
-        if flags.automatedInstall and self.data.reboot.action not in [KS_REBOOT, KS_SHUTDOWN]:
-            self._reboot = False
+def create_etc_symlinks():
+    """I don't know why I live. Maybe I should be killed."""
```
I really like this doc string, however I don't understand the meaning :(

jkonecny12 commented on pyanaconda/rescue.py:
```diff
@@ -182,105 +406,154 @@ def input(self, args, key):
+        return not self._rescue.get_locked_device_names()
+
+    def apply(self):
+        """Move along home."""
```
Maybe I'm just dumb but I don't get it? Am I missing something?

rvykydal replied:
I'll remove it, I don't understand it either.

jkonecny12 commented on pyanaconda/rescue.py:
```diff
@@ -182,105 +406,154 @@ def input(self, args, key):
 
-        self._root = None
+    def input(self, args, key):
+        """Move along home."""
```
The same here.

dedekind commented:
Hi, is there a chance to have this in F27?

jkonecny12 commented:
The F27 Anaconda is still not branched so this will be in F27.
